### ISSUE TITLE
snake_case `(invalid-name)`  for single letter variables in classes

### ISSUE THREAD

**Issue by definitely-not-a-furry** [PyCQA/pylint#8309]
### Bug description
```
#In this script variable 'f' in class 'SampleClass' is marked as '(invalid-name)':

#pylint: disable=missing-module-docstring
#pylint: disable=too-few-public-methods
class SampleClass():
    "SampleClass"
    def __init__(self):
        f = 'Hello world!'
        print(f)
```

### Command used

```shell
python -m pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:6:8: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.50/10
```

### Expected behavior

I'd expect no errors

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
```

### OS / Environment

(using $PSVersionTable)
PSVersion:                      7.3.2
PSEdition:                      Core
GitCommitId:                    7.3.2
OS:                             Microsoft Windows 10.0.22621
Platform:                       Win32NT
PSCompatibleVersions:           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion:      2.3
SerializationVersion:           1.1.0.1
WSManStackVersion:              3.0

### Additional dependencies

altgraph==0.17.3
astroid==2.14.2
certifi==2022.9.24
cffi==1.15.1
charset-normalizer==2.1.1
colorama==0.4.6
contourpy==1.0.6
cycler==0.11.0
decorator==4.4.2
dill==0.3.6
fonttools==4.38.0
future==0.18.2
idna==3.4
imageio==2.24.0
imageio-ffmpeg==0.4.8
isort==5.12.0
kiwisolver==1.4.4
lazy-object-proxy==1.9.0
matplotlib==3.6.2
mccabe==0.7.0
moviepy==1.0.3
numexpr==2.8.4
numpy==1.23.5
packaging==21.3
pandas==1.5.1
pandastable==0.13.0
pefile==2022.5.30
Pillow==9.3.0
platformdirs==3.0.0
proglog==0.1.10
py-wasapi-client==1.1.0
pycparser==2.21
pyee==9.0.4
pyinstaller==5.7.0
pyinstaller-hooks-contrib==2022.14
pylint==2.16.2
pynput==1.7.6
pyparsing==3.0.9
PySimpleGUI==4.60.4
python-dateutil==2.8.2
python-ffmpeg==1.0.16
PythonVideoConverter==1.0.3
pytz==2022.6
pywin32-ctypes==0.2.0
requests==2.28.1
six==1.16.0
SoundCard==0.4.2
termcolor==2.2.0
tomlkit==0.11.6
tqdm==4.64.1
typing_extensions==4.4.0
urllib3==1.26.12
windows-curses==2.3.1
wrapt==1.14.1
xlrd==2.0.1

**Comment by avylove** [comment]
This looks like the expected behavior to me. If I recall the default naming convention requires variable names to be 3 characters or longer. This is do encourage descriptive names, and therefore readability. You can adjust this using one of the options defined [here](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/invalid-name.html#cmdoption-variable-rgx) or add exceptions using [``--good-names``](https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#good-names) or [``--good-names-rgxs``](https://pylint.readthedocs.io/en/latest/user_guide/configuration/all-options.html#good-names-rgxs).

**Comment by mbyrnepr2** [comment]
A variable named "f" will produce:
Inside a class: `Variable name "f" doesn't conform to snake_case naming style (invalid-name)`.
Module-level: `Constant name "f" doesn't conform to UPPER_CASE naming style (invalid-name)`.

This is expected and @avylove has given some good advice how to deal with it if it doesn't suit a particular project's style; for example, physicists may wish to define `e = mc^2` etc. without the linter emitting a message in that case.

Having said that, I agree that the wording of the above messages don't give an obvious indication that the error is caused by the number of characters in the variable; perhaps we can add a note in the documentation to make that clearer - perhaps extend the good/bad example [here](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html)?

**Comment by Pierre-Sassoulas** [comment]
Closing as duplicate of https://github.com/PyCQA/pylint/issues/7305
